test/src/kbasesearchengine/test/main/IndexerCoordinatorTest.java

```diff
@@ -872,4 +885,36 @@ public void logDelayedWithCacheExpiry() throws Exception {
         verify(logger, never()).logError(any(String.class));
         verify(logger, never()).logError(any(Throwable.class));
     }
+    
+    @Test
+    public void internalShutdown() throws Exception {
+        final StatusEventStorage storage = mock(StatusEventStorage.class);
+        final LineLogger logger = mock(LineLogger.class);
+        final ScheduledExecutorService executor = mock(ScheduledExecutorService.class);
+        final SignalMonitor sm = new SignalMonitor();
+        
+        final IndexerCoordinator coord = new IndexerCoordinator(storage, logger, sm, 3, executor,
+                Arrays.asList(1), ST, SC);
+        
+        when(storage.get(StatusEventProcessingState.PROC, 3))
+                .thenThrow(new FatalRetriableIndexingException("whee"));
+        
```

Comment:
This test doesn't seem right. Looks like the 'when' clause should come above the coordinator in order to throw the exception during coord construction (I am assuming this is what you want). But even if you did that, coord would be null because of the exception thrown in the constructor and coord.awaitShutdown() would throw a null pointer.

Reply:
No, it gets thrown here, after construction: https://github.com/kbase/KBaseSearchEngine/blob/develop/lib/src/kbasesearchengine/main/IndexerCoordinator.java#L238

I purposely left it after the constructor so it wouldn't get thrown on construct.

Reply:
Actually, looking at this more closely, the when() isn't needed at all since the signal monitor is passed in and that's how the signal is triggered.

Reply:
Then you need coord.startIndexer() after the 'when' clause in order for control to reach loadEventsIntoQueue()?

Reply:
I just took out the when() clause and added the SignalMonitor fixes.

Reply:
Then test method name is misleading. This works well - 

        when(storage.get(StatusEventProcessingState.PROC, 3))
                .thenThrow(new FatalRetriableIndexingException("whee"));

        coord.startIndexer();
        Thread.sleep(100);

WIth powermokito's verifyNew, the call to the exception constructor can also be verified. But this would require powermokito's jar to be included I think.

Reply:
I updated the test name to reflect what's actually being tested. The reason I didn't do a full test like the above is that IIUC unit tests are supposed to mock their dependencies (other than value classes) rather than relying on dependency instances and their internal state. Admittedly, SignalMonitor is about as simple a dependency as you can get...